test/_apm_server.js

```diff
@@ -56,15 +56,12 @@ function APMServer (agentOpts, mockOpts = { expect: [] }) {
       assert.strictEqual(req.method, 'POST', `Unexpected HTTP method: ${req.method}`)
       assert.strictEqual(req.url, '/intake/v2/events', `Unexpected HTTP url: ${req.url}`)
 
-      self.on('close', () => {
-        res.end()
-      })
-
```

Comment:
Reviewer note: we shouldn't only end the response from the APM server when we server.close(). Below we properly respond.